src/charm.py

```diff
@@ -1122,6 +1125,38 @@ def _on_debug_ssh_relation_changed(self, _: ops.RelationChangedEvent) -> None:
             state.runner_config.virtual_machine_resources,
         )
 
+    @catch_charm_errors
+    def _on_image_relation_changed(self, _: ops.RelationChangedEvent) -> None:
+        """Handle image relation changed event."""
+        state = self._setup_state()
+
+        if state.instance_type != InstanceType.OPENSTACK:
+            return
+        if not self._get_set_image_ready_status():
+            return
+
+        runner_manager = self._get_openstack_runner_manager(state)
+        # 2024/04/12: Should be flush idle.
```

Comment:
Nitpick but maybe this comment should be prefixed with `TODO:` (and the date be ISO `2024-04-12`)

Reply:
I'll also fix this as a follow up PR since the tests are pretty hard to make pass. Thank you for the review!